hack/kcp-edge.sh

```diff
@@ -0,0 +1,253 @@
+while (( $# > 0 )); do
+    if [ "$1" == "start" ]; then
+        install=1
+    elif [ "$1" == "stop" ]; then
+        install=0
+    elif [ "$1" == "-v" ]; then
+        verbosity=1
+    elif [ "$1" == "--user" ]; then
+        user_type=$2
+    fi 
+    shift
+done
+
+# Check if docker is running
+if ! docker ps > /dev/null
+then
+  echo "Docker Not running ...."
+  exit
+fi
+
+# Check go version
+go_version=`go version | { read _ _ v _; echo ${v#go}; }`
+
+function ver { printf "%03d%03d%03d%03d" $(echo "$1" | tr '.' ' '); }
+if [ $(ver $go_version) -lt $(ver 1.19) ]; then
+    echo "Update your go version"
+fi
+
+# Check if a given process name is running
+process_running() {
+  SERVICE="$1"
+  if pgrep -f "$SERVICE" >/dev/null
+  then
+      echo "running"
+  else
+      echo "stopped" 
+  fi
+}
+
+# Check if kcp is running
+if [ $(process_running kcp) != "running" ]
+then
+    echo "kcp is not running - please start it ...."
+    exit
+fi
+
+# Check mailbox-controller is already running
+if [ $(process_running mailbox-controller) == "running" ]
+then
+    echo "An older deployment of mailbox-controller is already running - terminating it ...."
+    pkill -f mailbox-controller
+fi
+
+# Check edge-scheduler is already running
+if [ $(process_running "scheduler -v 2") == "running" ]
+then
+    echo "An older deployment of edge-scheduler is already running - terminating it ...."
+    pkill -f "scheduler -v 2"
+fi
+
+# Check placement-translator is already running
+if [ $(process_running placement-translator) == "running" ]
+then
+    echo "An older deployment of placement-translator is already running - terminating it ...."
+    pkill -f placement-translator
+fi
+
+kubectl ws root &> /dev/null
+
+if [ $install -eq 0 ]; then 
+   if kubectl get Workspace "$espw_name" &> /dev/null; then
+      kubectl delete ws "$espw_name"
+      echo "Deleted workspace: $espw_name"
+   fi
+   echo "kcp edge stopped ....."
+   exit
+fi
+
+
+wait_for_process(){
+  status=$(process_running $1)
+  MAX_RETRIES=5
+  retries=0
+  status_code=0
+  while [ $status != "running" ]; do
+      if [ $retries -eq $MAX_RETRIES ]; then
+           status_code=1
+           break
+      fi
+
+      retries=$(( retries + 1 ))
+      sleep 3
+      status=$(process_running $1)
+  done
+  echo $status_code
+}
+
+
+#(7): Start the edge-mc controller
+echo "****************************************"
+echo "Started deploying kCP-EDGE infra ...."
+echo "****************************************"
+
+
+if kubectl get Workspace "$espw_name" &> /dev/null; then
+   echo "espw workspace already exists -- using it:"
+   kubectl ws "$espw_name"
+else 
+   if [ $verbosity == 1 ]; then
+        kubectl ws create "$espw_name" --enter
+
+        if [ $user_type == "dev" ]; then
+            kubectl apply -f  ../config/crds 
+            kubectl apply -f  ../config/exports
+            echo "Finished populate the espw with kcp edge crds and apiexports"
+
+        elif [ $user_type == "kit" ]; then
```

Comment:
(1) This script should local copies regardless of who is invoking it, for the sake of consistency.

(2) With #363 those local copies will be available.

Reply:
Fixed - using the local copies now